app/src/main/java/ca/bc/gov/bchealth/datasource/DataStoreRepo.kt

```diff
@@ -1,69 +1,83 @@
         * we need to change key for NEW_FEATURE so that new storage with false flag is created
         * for existing users. Ex: Change NEW_FEATURE to NEW_FEATURE_FEDERAL_PASS
         * */
-        val NEW_FEATURE = booleanPreferencesKey("NEW_FEATURE")
-        val RECENT_FORM_DATA = stringPreferencesKey("RECENT_FORM_DATA")
+        const val NEW_FEATURE = "NEW_FEATURE"
+        const val RECENT_FORM_DATA = "RECENT_FORM_DATA"
+        const val PASS_PHRASE = "RECORD"
     }
 
-    val isOnBoardingShown: Flow<Boolean> = context.dataStore.data.map { preference ->
-        preference[ON_BOARDING_SHOWN] ?: false
+    val isOnBoardingShown: Flow<Boolean> = flow {
+        emit(encryptedPreferences.getBoolean(ON_BOARDING_SHOWN, false))
     }
 
-    suspend fun setOnBoardingShown(shown: Boolean = true) = context.dataStore.edit { preference ->
-        preference[ON_BOARDING_SHOWN] = shown
+    fun setOnBoardingShown(shown: Boolean = true) {
+        encryptedPreferences.edit().putBoolean(
+            ON_BOARDING_SHOWN, shown
+        ).apply()
     }
 
-    val isAnalyticsEnabled: Flow<Boolean> = context.dataStore.data.map { preference ->
-        preference[IS_ANALYTICS_ENABLED] ?: true
+    val isAnalyticsEnabled: Flow<Boolean> = flow {
+        emit(encryptedPreferences.getBoolean(IS_ANALYTICS_ENABLED, true))
     }
 
-    suspend fun trackAnalytics(shown: Boolean = true) = context.dataStore.edit { preference ->
-        preference[IS_ANALYTICS_ENABLED] = shown
+    fun trackAnalytics(shown: Boolean = true) {
+        encryptedPreferences.edit().putBoolean(
+            IS_ANALYTICS_ENABLED, shown
+        ).apply()
     }
 
-    val isNewFeatureShown: Flow<Boolean> = context.dataStore.data.map { preference ->
-        preference[NEW_FEATURE] ?: false
+    val isNewFeatureShown: Flow<Boolean> = flow {
+        emit(
+            encryptedPreferences.getBoolean(NEW_FEATURE, false)
+        )
     }
 
-    suspend fun setNewFeatureShown(shown: Boolean = true) = context.dataStore.edit { preference ->
-        preference[NEW_FEATURE] = shown
+    fun setNewFeatureShown(shown: Boolean = true) {
+        encryptedPreferences.edit().putBoolean(
+            NEW_FEATURE, shown
+        ).apply()
     }
 
-    val isRecentFormData: Flow<String> = context.dataStore.data.map { preference ->
-        preference[RECENT_FORM_DATA] ?: ""
+    val isRecentFormData: Flow<String> = flow {
+        emit(encryptedPreferences.getString(RECENT_FORM_DATA, "") ?: "")
     }
 
-    suspend fun setRecentFormData(formData: String) = context.dataStore.edit { preference ->
-        preference[RECENT_FORM_DATA] = formData
+    fun setRecentFormData(formData: String) {
+        encryptedPreferences.edit().putString(
+            RECENT_FORM_DATA, formData
+        ).apply()
+    }
+
+    val getPassPhrase: ByteArray = encryptedPreferences
+        .getString(PASS_PHRASE, "")?.toByteArray() ?: "".toByteArray()
+
+    fun setPassPhrase(passPhrase: ByteArray) {
```

Comment:
I thought getters/setters weren't recommended in Kotlin. Could you use properties?

Reply:
Yes, properties are used for the above-mentioned code. For the rest of the getters and setters methods, the getter and setter object types are different. Therefore properties cannot be used for other preferences in this class.